### PR TITLE
Delete /v3/company/* endpoints

### PR DESCRIPTION
**Pull request by alixedi**
### Description of change

This removes:

* `/v3/company` endpoints:

```
/v3/company
/v3/company/<uuid:pk> 
/v3/company/<uuid:pk>/archive 
/v3/company/<uuid:pk>/audit 
/v3/company/<uuid:pk>/one-list-group-core-team 
/v3/company/<uuid:pk>/timeline 
/v3/company/<uuid:pk>/unarchive
```

* `test_company_views_v3.py`
* `CompanyViewSetV3`
* `CompanySerializerV3`
* `BaseCompanyViewSet` (Because it was left with a single subclass - `CompanyViewSetV4`)
* `BaseCompanySerializer` (Because it was left with a single subclass - `CompanySerializerV4`)

Some code needed to be upgraded to the v4 api:

* Timeline tests: https://github.com/uktrade/data-hub-leeloo/pull/1676/commits/92be6458e3ac3d1f5b8257bf3bc06aec6d1d3767
* `update_company_global_hq` command: https://github.com/uktrade/data-hub-leeloo/pull/1676/commits/2a6b430ea893089610e0fb037ff9e79378013b2a

### Checklist

* [x] Has a new newsfragment been created? Check [changelog/README.rst](https://github.com/uktrade/data-hub-leeloo/blob/master/changelog/README.rst) for instructions
* [ ] Have any relevant search models been updated?
* [ ] Have any relevant fixtures (`fixtures/test_data.yaml`) been updated?
* [ ] Have any relevant select-/prefetch-related field lists in the views and search apps been updated?
* [ ] Has the admin site been updated (for new models, fields etc.)?
* [ ] Has the README been updated (if needed)?

### CODE REVIEW

File: datahub/company/views.py [hunk 1]
```diff
@@ -38,9 +37,10 @@
 from datahub.oauth.scopes import Scope
 
 
-class BaseCompanyViewSet(ArchivableViewSetMixin, CoreViewSet):
-    """Base Company view set."""
+class CompanyViewSetV4(ArchivableViewSetMixin, CoreViewSet):
```
Comment (marcofucci) [inline datahub/company/views.py]: I think this can become `CompanyViewSet` as there's just one version

File: datahub/company/views.py [hunk 2]
```diff
@@ -38,9 +37,10 @@
...
+    """Company view set V4."""
```
Comment (marcofucci) [inline datahub/company/views.py]: `V4` can go I think as there's just one version

File: datahub/company/views.py [hunk 3]
```diff
@@ -38,9 +37,10 @@
...
 
+    serializer_class = CompanySerializerV4
```
Comment (marcofucci) [inline datahub/company/views.py]: I think `CompanySerializerV4` can become `CompanySerializer` as there's just one version now

File: changelog/company/v3_company_endpoints.removal.rst [hunk 4]
```diff
@@ -0,0 +1,9 @@
+The ``/v3/company`` endpoints have been removed. These include:
+
+/v3/company
```
Comment (reupen) [inline changelog/company/v3_company_endpoints.removal.rst]: Check the formatting here – they seem to be formatting on one line.

File: datahub/company/urls/company.py [hunk 5]
```diff
@@ -65,20 +43,6 @@
...
-    path(
-        'company/<uuid:pk>/one-list-group-core-team',
-        one_list_group_core_team,
-        name='one-list-group-core-team',
-    ),
-]
-
 urls_v4 = [
     path('company', company_collection_v4, name='collection'),
     path('company/<uuid:pk>', company_item_v4, name='item'),
```
Comment (marcofucci) [inline datahub/company/urls/company.py]: I think that all the v4 suffix in this module can go as there's only one version now, including `urls_v4`

File: datahub/company/serializers.py [hunk 6]
```diff
@@ -282,6 +282,21 @@ class BaseCompanySerializer(PermittedFieldsModelSerializer):
     have to enter company numbers for UK establishments manually.
     """
 
+    registered_address = AddressSerializer(
```
Comment (reupen) [inline datahub/company/serializers.py]: Would be good to move `registered_address` and `address` further down so that they are next to the other fields.

**Pull request merged**